sycl-jit/jit-compiler/lib/rtc/DeviceCompilation.cpp

```diff
@@ -96,23 +103,20 @@ struct GetLLVMModuleAction : public ToolAction {
 
 } // anonymous namespace
 
-llvm::Expected<std::unique_ptr<llvm::Module>>
+Expected<std::unique_ptr<llvm::Module>>
 jit_compiler::compileDeviceCode(InMemoryFile SourceFile,
                                 View<InMemoryFile> IncludeFiles,
                                 View<const char *> UserArgs) {
   const std::string &DPCPPRoot = getDPCPPRoot();
   if (DPCPPRoot == InvalidDPCPPRoot) {
-    return llvm::createStringError("Could not locate DPCPP root directory");
+    return createStringError("Could not locate DPCPP root directory");
   }
 
   SmallVector<std::string> CommandLine = {"-fsycl-device-only"};
-  // TODO: Allow instrumentation again when device library linking is
-  //       implemented.
-  CommandLine.push_back("-fno-sycl-instrument-device-code");
```

Comment:
NB: This PR is tested by the existing E2E, which now works with device instrumentation enabled.